contrib/pcap2emulation.py

```diff
@@ -0,0 +1,361 @@
+                            int(layers["usb_usb_bmRequestType_recipient"], 16)
+                        )
+                        s += ",Request=0x{:02x}".format(
+                            int(layers["usb_usb_setup_bRequest"])
+                        )
+                        s += ",Value=0x{:04x}".format(
+                            int(layers["usb_usb_setup_wValue"], 16)
+                        )
+                        s += ",Idx=0x{:04x}".format(int(layers["usb_usb_setup_wIndex"]))
+                        if "usb_usb_data_fragment" in layers:
+                            data = layers["usb_usb_data_fragment"].replace(":", "")
+                        else:
+                            data = "00" * int(layers["usb_usb_setup_wLength"])
+                        s += ",Data={}".format(
+                            str(base64.b64encode(bytes.fromhex(data)), "utf-8")
+                        )
+                        s += ",Length=0x{:x}".format(
+                            int(layers["usb_usb_setup_wLength"])
+                        )
+                        event = {"Id": s}
+
+                        if direction:
+                            # Duplicate the outgoing data as response and add the event to the device events
+                            event["Data"] = str(
+                                base64.b64encode(bytes.fromhex(data)), "utf-8"
+                            )
+                            self._save_event(event)
+
+                    elif "usb_usb_control_Response" in layers:
+                        # Found CONTROL URB response
+                        data = layers["usb_usb_control_Response"].replace(":", "")
+                        event["Data"] = str(
+                            base64.b64encode(bytes.fromhex(data)), "utf-8"
+                        )
+
+                        # now that the event is complete it can be added to the device events
+                        self._save_event(event)
+
+                    # TODO: Add missing event GetCustomIndex
+
+                elif get_int(layers["usb"]["usb_usb_transfer_type"]) == URB_BULK:
+                    # TODO: check it
+                    event = self._get_transfer_event("BulkTransfer", layers)
+                    if event:
+                        self._save_event(event)
+
+                else:
+                    if not self.device:
+                        continue
+
+                    print(
+                        "Unknown frame type: " + layers["usb"]["usb_usb_transfer_type"]
+                    )
+
+        # Save the last USB device
+        if "UsbInterfaces" in self.device:
+            self._save_phase()
+
+
+if __name__ == "__main__":
+    if {"-?", "-h", "--help", "--usage"}.intersection(set(sys.argv)):
+        usage(0)
+
+    if len(sys.argv) < 4:
+        usage(1)
+
+    path = os.path.abspath(os.path.expanduser(os.path.expandvars(sys.argv[1])))
+    parser = Pcap2Emulation(sys.argv[2], sys.argv[3:])
```

Comment:
why not use `argparse`?  It's a lot more flexible to expand arguments later and re-order and such.